boot/rotate-session-secret.sh

```diff
@@ -0,0 +1,324 @@
+GENERATE_SECRET_SCRIPT="$BOOT_ROOT/generate-session-secret.js"
+
+function usage() {
+    cat << EOF
+Usage: $0 [options]
+
+Rotate JWT session secret with grace period support.
+
+Options:
+    -d, --dry-run           Show what would be done without making changes
+    -f, --force             Force rotation even if recent rotation detected
+    -g, --grace-period SEC  Grace period in seconds (default: 86400)
+    -h, --help              Show this help message
+
+Environment Variables:
+    GRACE_PERIOD           Default grace period in seconds
+    DRY_RUN               Set to 'true' for dry run mode
+    FORCE                 Set to 'true' to force rotation
+
+Examples:
+    # Standard rotation with 24-hour grace period
+    $0
+    
+    # Dry run to see what would happen
+    $0 --dry-run
+    
+    # Forced rotation with custom grace period
+    $0 --force --grace-period 3600
+EOF
+    exit 1
+}
+
+function log() {
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] $*" >&2
+}
+
+function get_current_secret() {
+    mdata-get "sdc:application_metadata.$1" 2>/dev/null || echo ""
+}
+
+function set_sapi_metadata() {
+    local key="$1"
+    local value="$2"
+    
+    if [[ "$DRY_RUN" == "true" ]]; then
+        log "DRY RUN: Would set $key = $value"
+        return 0
+    fi
+    
+    log "Setting $key in SAPI metadata..."
+    if ! $BOOT_ROOT/set-sapi-metadata.sh "$key" "$value"; then
+        log "Warning: SAPI metadata update failed for $key (this may be expected in dev environments)"
+    else
+        log "Successfully set $key in SAPI metadata"
+    fi
+    return 0
+}
+
+function generate_new_secret() {
+    local secret
+    secret=$("$GENERATE_SECRET_SCRIPT")
+    
+    if [[ -z "$secret" ]]; then
+        log "ERROR: Failed to generate new secret"
+        exit 1
+    fi
+    
+    echo "$secret"
+}
+
+function generate_key_id() {
+    local timestamp
+    local random
+    
+    timestamp=$(date +%Y%m%d-%H%M%S)
+    random=$(openssl rand -hex 4)
+    
+    echo "key-${timestamp}-${random}"
+}
+
+function check_rotation_needed() {
+    local current_secret
+    local current_key_id
+    local last_rotation
+    local now
+    local rotation_age
+    local min_interval
+    
+    # Check if this is a manual rotation request
+    if [[ "$FORCE" == "true" ]]; then
+        log "Force flag set, skipping all rotation checks"
+        return 0
+    fi
+    
+    # Check if secrets are already configured (prevents rotation on boot)
+    current_secret=$(get_current_secret "SESSION_SECRET_KEY")
+    current_key_id=$(get_current_secret "SESSION_SECRET_KEY_ID")
+    last_rotation=$(get_current_secret "SESSION_SECRET_ROTATION_TIME")
+    
+    if [[ -n "$current_secret" && -n "$current_key_id" ]]; then
+        log "Secrets are already configured:"
+        log "  Key ID: $current_key_id"
+        if [[ -n "$last_rotation" ]]; then
+            log "  Last rotation: $(date -d @$last_rotation 2>/dev/null || echo $last_rotation)"
```

Comment:
Per shellcheck:
```
[Line 119:](javascript:setPosition(119, 46))
            log "  Last rotation: $(date -d @$last_rotation 2>/dev/null || echo $last_rotation)"
                                             ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.
>>                                                                              ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.
```